solarforecastarbiter/reference_forecasts/forecast.py

```diff
@@ -150,52 +135,31 @@ def resample(arg, freq='1h', closed=None):
     if arg is None:
         return None
     else:
-        return arg.resample(freq, closed=closed).mean()
-
+        return arg.resample(freq, label=label, closed=label).mean()
 
-def interpolate_args(*args, freq='15min'):
-    """Interpolate all positional arguments, allowing for None.
-
-    Parameters
-    ----------
-    *args : list of pd.Series or None
 
-    Returns
-    -------
-    list of pd.Series or None
+def reindex_fill_slice(arg, freq='5min', label=None, start=None, end=None,
+                       start_slice=None, end_slice=None,
+                       fill_method='interpolate'):
+    """Reindex data to shorter intervals (create NaNs) from `start` to
+    `end`, fill NaNs in gaps using `fill_method`, fill NaNs before first
+    valid time using bfill, fill NaNs after last valid time using ffill,
```

Comment:
is there any way a nan will be left in the middle after the application of `fill_method` that will then be ffilled?

Reply:
Maybe but I don't see how that's possible. Only times before the first non-nan and after the last non-nan would remain nan.